kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/service/SpecManagementServiceImpl.java

```diff
@@ -112,9 +112,9 @@ public synchronized void updateContainerSpec(final String serverTemplateId, fina
         serverTemplate.deleteContainerSpec(currentVersion.getId());
         serverTemplate.addContainerSpec(containerSpec);
 
-        templateStorage.update( serverTemplate );
+        templateStorage.update(serverTemplate);
 
-        notificationService.notify( new ServerTemplateUpdated( serverTemplate ) );
+        notificationService.notify(new ServerTemplateUpdated(serverTemplate));
         // in case container was started before it was update or update comes with status started update container in running servers
         if (currentVersion.getStatus().equals(KieContainerStatus.STARTED) || containerSpec.getStatus().equals(KieContainerStatus.STARTED)) {
```

Comment:
The test failure is related to this condition. Test expects that if you update container spec's status from STOP to START then controller will send create container request to Kie servers.
@karreiro I think that this condition should be splitted. If former spec was in state STOPPED and newer is in state STARTED then kieServerInstanceManager would start the container. If former spec was in state STARTED and newer is in state STARTED then kieServerInstanceManager would upgrade the container.

Reply:
@sutaakar In fact, the test was failing because the previous behavior was `startAndupgrade` the container, and I unintentionally changed it to only `upgrade`. I already applied the fix here, and everything is fine now.